classification/class_task.py

```diff
@@ -88,10 +88,82 @@ def evaluate(self, test_loader):
                 total += self.batch_size
                 correct += (preds == y).sum().item()
         self.acc = correct / total
-        print(f'Validation accuracy: {self.acc}')
-        
+        print(f'{self.label} validation accuracy: {self.acc}')    
 
     def log_progress(self, step):
         write_weight_histograms(self.writer, self.net, step)
         write_loss_scalars(self.writer, self.loss_info, step)
         write_acc(self.writer, self.acc, step)
+
+
+class MLP_Classification():
+    def __init__(self, label, parameters):
+        super().__init__()
+        self.writer = SummaryWriter(comment=f"_{label}_training")
+        self.label = label
+        self.lr = parameters['lr']
+        self.hidden_units = parameters['hidden_units']
+        self.mode = parameters['mode']
+        self.batch_size = parameters['batch_size']
+        self.num_batches = parameters['num_batches']
+        self.x_shape = parameters['x_shape']
+        self.classes = parameters['classes']
+        self.save_model_path = f'{parameters["save_dir"]}/{label}_model.pt'
+        self.best_acc = 0.
+        self.dropout = parameters['dropout']
+        self.init_net(parameters)
+    
+    def init_net(self, parameters):
+        if not os.path.exists(parameters["save_dir"]):
+            os.makedirs(parameters["save_dir"])
+
+        model_params = {
+            'input_shape': self.x_shape,
+            'classes': self.classes,
+            'batch_size': self.batch_size,
+            'hidden_units': self.hidden_units,
+            'mode': self.mode,
+            'dropout': self.dropout
+        }
+        if self.dropout:
+            self.net = MLP_Dropout(model_params).to(DEVICE)
+        else:
+            self.net = MLP(model_params).to(DEVICE)
+        self.optimiser = torch.optim.Adam(self.net.parameters(), lr=self.lr)
```

Comment:
not too fussed about this, but if we are comparing to vanilla SGD, it would be more accurate to use SGD optimiser for training.

Reply:
Oh yeah, good point, will fix. Should we be using SGD for the Bayesian models then as well?

Reply:
Both SGD and ADAM are SGD techniques but ADAM is adaptive. I think it should be fine if we still use ADAM for Bayesian models? But for MLP, it is a bit misleading if we claim to be using SGD.

Reply:
I'm happy to leave the BNNs with adam and change the MLPs to SGD. @Eli-Persky also happy with this?

Reply:
@tennisonliu  still with the lr scheduler, or remove that also?

Reply:
the lr scheduler works with any optimiser! so we can keep as-is?

Reply:
is it still "vanilla SGD"?

Reply:
yea i reckon its fine